prow/cmd/deck/job_history.go

```diff
@@ -346,15 +346,23 @@ func getBuildData(ctx context.Context, bucket storageBucket, dir string) (buildD
 		return b, fmt.Errorf("failed to read started.json: %v", err)
 	}
 	b.Started = time.Unix(started.Timestamp, 0)
-	if commitHash, err := getPullCommitHash(started.Pull); err == nil {
-		b.commitHash = commitHash
-	}
 	finished := gcs.Finished{}
 	err = readJSON(ctx, bucket, path.Join(dir, prowv1.FinishedStatusFile), &finished)
 	if err != nil {
 		b.Result = "Pending"
+		for _, ref := range started.Repos { 
+			if strings.Contains(ref, "," + started.Pull + ":"){
+				started.Pull = ref
```

Comment:
Im not completely happy with this approch, `started.Pull `field only has the PR number when the build is in pending state thus I decided to the Repos field from started.json.

Reply:
Here is a reference to the conical source of truth for these fields. I would make sure these fields stay consistent.

Reply:
Thanks for the heads up with that file @MushuEE , I've reviewed it and the approach seems to be consistent with it, that why I iterate the repository to find the reference instead of ussing the first one, just in case the order changes somehow.

Reply:
Would also to see unit test coverage for this scenario